src/qt/sendcoinsdialog.cpp

```diff
@@ -828,3 +829,33 @@ void SendCoinsDialog::coinControlUpdateLabels()
         ui->labelCoinControlInsuffFunds->hide();
     }
 }
+
+SendConfirmationDialog::SendConfirmationDialog(const QString &title, const QString &text,
+    int msecDelay, QWidget *parent) :
+    QMessageBox(QMessageBox::Question, title, text, QMessageBox::Yes | QMessageBox::Cancel, parent),
+    msecDelay(msecDelay)
+{
+    setDefaultButton(QMessageBox::Cancel);
+}
+
+int SendConfirmationDialog::exec()
+{
+    button(QMessageBox::Yes)->setEnabled(false);
+    QTimer::singleShot(msecDelay, this, SLOT(enableSendButton()));
```

Comment:
@jonasschnelli, I don't think it's possible to invalidate a singleShot. There's never any timer created for me to invalidate. It functions like the static QMessageBox::info/warning/etc functions.

I would expect that Qt planned for the possibility of the connected object disappearing, seeing as it could happen often. A quick google turns up that signals are automatically disconnected: http://stackoverflow.com/questions/10570857/are-signals-in-qt-automatically-disconnected-when-one-of-the-class-is-deleted